### PR TITLE
[webview_flutter] webview load HTML string

### PR DESCRIPTION
**Pull request by truongsinh**
Sample screenshot code
```dart
const WebView(
              htmlString: """
<u><em><strong>You can do HTML too!</strong></em></u><br />
<img src="https://upload.wikimedia.org/wikipedia/commons/1/17/Google-flutter-logo.png" style="float: left">
Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.
<img src="https://upload.wikimedia.org/wikipedia/commons/1/17/Google-flutter-logo.png" style="float: right"> 
Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
              """,
            ),
```
will yield
screen shot 2019-03-05 at 21 55 23

### CODE REVIEW

File: packages/webview_flutter/lib/webview_flutter.dart [hunk 1]
```diff
@@ -112,6 +112,7 @@ class WebView extends StatefulWidget {
     Key key,
     this.onWebViewCreated,
     this.initialUrl,
+    this.htmlString,
```
Comment (mehmetf) [inline packages/webview_flutter/lib/webview_flutter.dart]: Can you assert that either htmlString or initialUrl must be set (but not both).

**Comment by amirh** [comment]
Haven't reviewed the PR yet, just a comment on the public API:
The html string is an edge trigger and cannot model well as a widget parameter(the API will hit issues similar to the ones described here: https://medium.com/flutter-io/api-design-tip-for-complex-flutter-widgets-7794e9c7144c ).

I'd suggest we start just with a controller API.

**Comment by truongsinh** [comment]
@amirh @mehmetf, so to sum up:
- level trigger (parameter) name changed from `htmlString` to `initialHtmlString`
  - assert that either `initialHtmlString` or `initialUrl` must be set, but not both **warning this can be a breaking API, as `initialUrl` has not been required**
- expose edge trigger / controller API `webViewController.loadHtmlString` as well

On that breaking API note, should we separate `WebView`'s constructor into 3:
- `WebView` ==> basically no `initalUrl` nor `initialHtmlString`, setting `WebView(initalUrl: "")` should be warned as deprecated
- `WebView.withInitialUrl`
- `WebView.withInitialHtmlString`

to have the API more type-safe, and more declarative?

File: packages/webview_flutter/test/webview_flutter_test.dart [hunk 2]
```diff
@@ -33,6 +33,19 @@ void main() {
     await tester.pumpWidget(const WebView());
   });
 
+  testWidgets('HTML string', (WidgetTester tester) async {
```
[the remark this answers is not in the document]
Comment (truongsinh) [reply packages/webview_flutter/test/webview_flutter_test.dart]: Thanks, I believe e2e test was not available when this PR is open, I will revisit it.

[the remark this answers is not in the document]
Comment (amirh) [reply packages/webview_flutter/test/webview_flutter_test.dart]: yes, it's very new

**Comment by tuanp** [comment]
@truongsinh Can you try to loadURL with [Data URLs](https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/Data_URIs)?
webViewController.loadUrl(Uri.dataFromString(
                                    yourHtmlString,
                                    mimeType: 'text/html',
                                    encoding: utf8)
                                .toString());

**Comment by amirh** [comment]
I filed https://github.com/flutter/flutter/issues/32827 to track this feature.
This PR needs some work before it can be landed (followup on comments I left before) as it seems to be stale for more than a month now I'm going to close this PR, feel free to re-open if you get a chance to iterate and want me to take a look again.

I mentioned this PR on https://github.com/flutter/flutter/issues/32827 so that someone looking to fix that issue have the option to pick up this work form where you left it.

Thanks!

**Pull request closed without merging**